server/gfmd/user.c

```diff
@@ -564,8 +564,15 @@ user_auth_add_one(void *closure, struct db_user_auth_arg *p)
 			gflog_warning(GFARM_MSG_UNFIXED,
 			"user_auth_add_one: %s", gfarm_error_string(e));
 	}
-}
 
+	/* free temporary info */
```

Comment:
Please remove the following line just before the user_auth_add_one() function:

	struct db_user_auth_arg;

Instead, do the following at the first part of user.c:

	#include "db_common.h"

Also, probably it's better to change the following comment

	/* free temporary info */

to

	/*
	 * unlike user_add_one(), the memory owner of *p is not changed to user.c
	 * thus, it must be freed here.
	 */

Because the "free temporary info" message doesn't give enough additional information.

Note that the user_add_one() function has the following comment:

	/* The memory owner of `*ui' is changed to user.c */